### PR TITLE
commander: ensure health report is always sent out before failsafe notificaation

### PR DESCRIPTION
**Pull request by bkueng**
As the failsafe message can reference the health report, the health report needs to be sent out first. This is generally the case, except there is a rate limiter set to 2 seconds. So if the report changes quickly, it is sent out delayed (potentially after the failsafe report).

### CODE REVIEW

File: src/modules/commander/HealthAndArmingChecks/Common.cpp [hunk 1]
```diff
@@ -317,3 +317,12 @@ bool Report::report(bool is_armed, bool force)
 			       current_results.health.error, current_results.health.warning);
 	return true;
 }
+
+bool Report::reportIfUnreportedDifferences(bool is_armed)
+{
+	if (_had_unreported_difference) {
+		return report(is_armed, true);
```
Comment (MaEtUgR) [inline src/modules/commander/HealthAndArmingChecks/Common.cpp:324]: Mildly related but where's `is_armed` even used in report()? 🤔

**Review by MaEtUgR — DISMISSED** [review]
Thanks for looking itno it! A classic C callback setup to allow triggering a healt_and_arming_checks action based on a consition in the completely separate failsafe state machine.

It works and crosses the nice architectural separation to have the nice behavior to only update in the necessary case.

File: src/modules/commander/HealthAndArmingChecks/Common.cpp (hunk 1 above)
Comment (bkueng) [reply src/modules/commander/HealthAndArmingChecks/Common.cpp]: You are right, it's not used. I removed it in the 2. commit.

**Review by sfuhrer — DISMISSED** [review]

**Review by sfuhrer — COMMENTED** [review]
One CI failure is real: https://github.com/PX4/PX4-Autopilot/actions/runs/11934542772/job/33263873602?pr=23973

**Pull request merged**